torch/optim/adagrad.py

```diff
@@ -40,6 +40,8 @@ def __init__(self, params, lr=1e-2, lr_decay=0, weight_decay=0, initial_accumula
 
         for group in self.param_groups:
             for p in group['params']:
+                if isinstance(p, torch.nn.parameter.UninitializedParameter):
+                    raise ValueError('torch.optim.AdaGrad does not support torch.nn.parameter.UninitializedParameter')
```

Comment:
nit: You can add: "Make sure to run the dummy forward before creating the optimizer."

Reply:
yes, good point!